Plugin doesn't perform InsideOut for canUnify

When investigating https://github.com/isovector/polysemy-zoo/issues/21, I learned that the new 0.2.0.1 plugin logic is subtly broken:

```haskell
/home/sandy/prj/polysemy-zoo/test/MTLSpec.hs:75:11: error:                                                       
    • Ambiguous use of effect 'Writer'                                                                           
      Possible fix:                
        add (Member (Writer [a0]) '[Writer [Int]]) to the context of 
          the type signature
```

Oopsies. 

https://github.com/isovector/polysemy/blob/ca0c68e1c35be4ec45662c36b7dbb12980d29e81/polysemy-plugin/src/Polysemy/Plugin/Fundep.hs#L86-L93

`canUnify` should recursively attempt to solve equalities.

--- a/polysemy-plugin/src/Polysemy/Plugin/Fundep.hs
+++ b/polysemy-plugin/src/Polysemy/Plugin/Fundep.hs
@@ -89,14 +89,14 @@ canUnify wanted given =
       (g, gs) = splitAppTys given
    in (&& eqType w g) . flip all (zip ws gs) $ \(wt, gt) ->
      or [ isTyVarTy wt
-        , eqType gt wt
+        , eqType wt gt
+        , canUnify wt gt
         ]
 
 
-
 mkWanted :: Bool -> CtLoc -> Type -> Type -> TcPluginM (Maybe Ct)
-mkWanted mustUnify loc wanted given =
-  if (not mustUnify || canUnify wanted given)
+mkWanted must_unify loc wanted given =
+  if (not must_unify || canUnify wanted given)
      then do
        (ev, _) <- unsafeTcPluginTcM $ runTcSDeriveds $ newWantedEq loc Nominal wanted given
        pure $ Just $ CNonCanonical ev

--- a/polysemy-plugin/test/PluginSpec.hs
+++ b/polysemy-plugin/test/PluginSpec.hs
@@ -10,6 +10,7 @@ import GHC.Exts
 import Polysemy
 import Polysemy.Error
 import Polysemy.State
+import Polysemy.Output
 import Test.Hspec
 
 
@@ -115,6 +116,13 @@ spec = do
       flipShouldBe (Right @String (10 :: Int, True))  . run $ runError $ runState 0 errState
     it "should interpret against Bool/Float" $ do
       flipShouldBe (Right @Bool (10 :: Float, True))  . run $ runError $ runState 0 errState
+
+  describe "Output effect" $ do
+    it "should unify recursively" $ do
+      -- TODO(sandy): This should unify even without the type app. Bug #95
+      flipShouldBe 11 . sum @[] . fst . run . runFoldMapOutput id $ do
+        output [1]
+        output $ replicate 2 5
 
 
   describe "Lift effect" $ do

